test/test_linalg.py

```diff
@@ -5689,28 +5689,76 @@ def run_test(matsize, batchdims, mat_chars):
 
     @skipCUDAIfNoMagma
     @skipCPUIfNoLapack
-    @dtypes(torch.double)
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
     def test_cholesky_inverse(self, device, dtype):
-        from torch.testing._internal.common_utils import random_symmetric_pd_matrix
-        a = random_symmetric_pd_matrix(5, dtype=dtype, device=device)
+        from torch.testing._internal.common_utils import random_hermitian_pd_matrix
 
-        # compute inverse directly
-        inv0 = torch.inverse(a)
+        def run_test(shape, batch, upper, contiguous):
+            A = random_hermitian_pd_matrix(shape, *batch, dtype=dtype, device=device)
+            if A.numel() > 0 and not contiguous:
+                A = A.transpose(-2, -1)
+                self.assertFalse(A.is_contiguous())
+            L = torch.linalg.cholesky(A)
+            expected_inverse = torch.inverse(A)
+            L = L.conj().transpose(-2, -1) if upper else L
+            actual_inverse = torch.cholesky_inverse(L, upper)
+            self.assertEqual(actual_inverse, expected_inverse)
 
-        # default case
-        chol = torch.cholesky(a)
-        inv1 = torch.cholesky_inverse(chol, False)
-        self.assertLessEqual(inv0.dist(inv1), 1e-12)
+        shapes = (0, 3, 5)
+        batches = ((), (0,), (3, ), (2, 2))
+        for shape, batch, upper, contiguous in list(itertools.product(shapes, batches, (True, False), (True, False))):
+            run_test(shape, batch, upper, contiguous)
 
-        # upper Triangular Test
-        chol = torch.cholesky(a, True)
-        inv1 = torch.cholesky_inverse(chol, True)
-        self.assertLessEqual(inv0.dist(inv1), 1e-12)
+        # check the out= variant
+        A = random_hermitian_pd_matrix(3, 2, dtype=dtype, device=device)
+        L = torch.linalg.cholesky(A)
 
-        # lower Triangular Test
-        chol = torch.cholesky(a, False)
-        inv1 = torch.cholesky_inverse(chol, False)
-        self.assertLessEqual(inv0.dist(inv1), 1e-12)
+        # batched column major out (memory is used in computations directly)
```

Comment:
Add a comment that these out= tests are necessary to exercise the different out codepaths (which test_ops.py does not cover), and enumerate each code path up front so the reader can follow the cases

Reply:
Added the following text
```python
        # There are two code paths currently for the out= variant
        # 1. When 'out' tensor is in Fortran (column-major) memory format
        # then the fast route is taken and the storage is reused directly in the computations
        # 2. When 'out' tensor is not in Fortran format then a temporary tensor is allocated internally
        # and the result is copied from the temporary tensor to 'out' tensor
```